go/vt/vtgate/planbuilder/simple_projection.go

```diff
@@ -23,31 +23,32 @@ import (
 	"vitess.io/vitess/go/vt/vtgate/engine"
 )
 
-var _ logicalPlan = (*subquery)(nil)
+var _ logicalPlan = (*simpleProjection)(nil)
 
-// subquery is a logicalPlan that wraps a subquery.
-// This primitive wraps any subquery that results
+// simpleProjection is used for wrapping a derived table.
+// This primitive wraps any derived table that results
 // in something that's not a route. It builds a
-// 'table' for the subquery allowing higher level
-// constructs to reference its columns. If a subquery
+// 'table' for the derived table allowing higher level
+// constructs to reference its columns. If a derived table
 // results in a route primitive, we instead build
 // a new route that keeps the subquery in the FROM
 // clause, because a route is more versatile than
-// a subquery.
-type subquery struct {
+// a simpleProjection.
+// this should not be used by the gen4 planner
+type simpleProjection struct {
 	logicalPlanCommon
 	resultColumns []*resultColumn
-	esubquery     *engine.Subquery
+	primitive     *engine.SimpleProjection
```

Comment:
nit: in other logical plans, we use `e` as a suffix to indicate it is an engine.Primitive. We can follow the same suite here.